src/Rector/Deprecation/FunctionToServiceRector.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Rector\ValueObject\FunctionToServiceConfiguration;
+use PhpParser\Node;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * Replaces deprecated function call with service method call.
+ *
+ * What is covered:
+ * - Static replacement
+ *
+ * Improvement opportunities
+ * - Dependency injection
+ */
+class FunctionToServiceRector extends AbstractRector implements ConfigurableRectorInterface
+{
+
+    /**
+     * @var FunctionToServiceConfiguration[]
+     */
+    private array $functionToServiceConfigs;
+
+    public function configure(array $configuration): void {
+        foreach ($configuration as $value) {
+            if (!($value instanceof FunctionToServiceConfiguration)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Each configuration item must be an instance of "%s"',
+                    FunctionToServiceConfiguration::class
+                ));
+            }
+        }
+
+        $this->functionToServiceConfigs = $configuration;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getNodeTypes(): array
+    {
+        return [
+            Node\Expr\FuncCall::class,
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function refactor(Node $node): ?Node
+    {
+        foreach ($this->functionToServiceConfigs as $configuration) {
+            /** @var Node\Expr\FuncCall $node */
+            if ($this->getName($node->name) === $configuration->getDeprecatedFunctionName()) {
+                // This creates a service call like `\Drupal::service('file_system').
+                // TODO use dependency injection.
```

Comment:
I don't know if this `TODO` is a copy over, but I am assuming it'd be near impossible to support dependency injection – is it a class that is a service? Then we have to update `services.yml`. Does it implement `ContainerInjectionInterface`? Then we have to modify `create`.

I think we can remove it because we'll never do it.

Reply:
Yeah it's a copy. But sure, can remove it while we're at it.